packages/nx-aws-cache/src/tasks-runner/aws-cache.ts

```diff
@@ -126,9 +119,15 @@ export class AwsCache implements RemoteCache {
   private async createAndUploadFile(hash: string, cacheDirectory: string): Promise<boolean> {
     try {
       const tgzFilePath = this.getTgzFilePath(hash, cacheDirectory);
-
       await this.createTgzFile(tgzFilePath, hash, cacheDirectory);
-      await this.uploadFile(hash, tgzFilePath);
+      const sourceFileStream = createReadStream(tgzFilePath);
+
+      if (this.encryptConfig) {
+        const encrypt = new Encrypt(this.encryptConfig);
+        await this.uploadFileMultipart(hash, sourceFileStream.pipe(encrypt));
```

Comment:
What is the reason that encrypted stream has to be uploaded as multipart? Why don't you use just `this.uploadFile()`?

Reply:
When upload file with transform stream you don't know what will be the final ContentLength,
so you have to upload that file as multipart upload

Reply:
How can I test this?

Reply:
Test what? In aws-cache.test.ts are e2e tests. 

In uploadFile you use PutObjectCommand. PutObjectCommand required ContentLength.
Encryption use stream transform. Stream transform do not set size of output  buffer, so for resolve this you have to use uploadmultipart upload which not require contentLength on start.

Reply:
When you will use normal uploadFile, minio will not save that file, because do not pass contentLength in headers.

Reply:
Otherwise we can create one uploadFile method but with multipart for each type of file.

Reply:
My new commit has one method uploadFile, without uploadFileMultipart.